### DISCUSSION TITLE
evtx2es re-broken: [Temporary Hackaround :)]

### DISCUSSION THREAD

**Discussion by zaphode** [Security-Onion-Solutions/securityonion#3822] [General] [answered]
evtx2es was working until I updated to 2.3..40. 
Is authentication to the Elastic Search API (port:9200) added in 2.3.40?  I can query **GET /_cat/indices** from in Dev Tools but get an empty reply if I try to run it from curl. It seems like it would be an access issue. 

What I have checked:

- Made sure access was open in so-allow
- tested simple query using curl (failed) 

 
![image](https://user-images.githubusercontent.com/3597404/114213914-42e5d700-9929-11eb-9f6f-1c72093e27d2.png)

**Comment by defensivedepth** [comment, marked as answer]
Please review https://github.com/Security-Onion-Solutions/securityonion/discussions/3449

You will need to connect via https now.

**Reply by zaphode** [reply]
I tried this but it looks like there is an authentication error. 
![image](https://user-images.githubusercontent.com/3597404/114427010-445d0c80-9b80-11eb-8c5b-b0411a0765cc.png)

**Reply by zaphode** [reply]
As it does work in curl using https with no authentication I also tried putting in a blank login. Is there maybe a login required to write to Elasticsearch in this version? 

![image](https://user-images.githubusercontent.com/3597404/114428656-137dd700-9b82-11eb-9c8a-f32f6cb57539.png)


![image](https://user-images.githubusercontent.com/3597404/114428567-fa752600-9b81-11eb-8066-10abfb10fa1a.png)

**Reply by defensivedepth** [reply]
That is a bug in evtx2es - I submitted a PR to fix it https://github.com/sumeshi/evtx2es/pull/6

(It was trying to authenticate using a username `elastic`, even though `--login` was not specified

**Reply by zaphode** [reply]
Good call, got it working

**Comment by zaphode** [comment]
If anyone else is using evtx2es, here is a temporary hackaround until it is fixed.

Edit the '/usr/local/bin/evtx2es' (or wherever you have this file stored).

1. Add an extra elif statement to check for the state (scheme == 'https') and (login == "") in the Elasticsearch() initialization.
![image](https://user-images.githubusercontent.com/3597404/114636405-8f644600-9c8c-11eb-9fc4-4ed15ab118e2.png)

2. In the argument parser (around Line 293) replace default="elastic" with default="" 
![image](https://user-images.githubusercontent.com/3597404/114636645-187b7d00-9c8d-11eb-9b6a-c08c69e85eb7.png)


Suppress the "Unverified HTTPS request" messages:
**export PYTHONWARNINGS="ignore:Unverified HTTPS request"**

Ex: 
- **`evtx2es --scheme https  --host localhost --index so-evtx2es --pipeline import.wel sysmon3_win10.evtx`**

Or better just hide the Unverified HTTPS request:
